_RELEASE/Packs/cube/Scripts/utils.lua

```diff
@@ -1,21 +1,552 @@
+		return -1;
+	elseif number > 0 then
+		return 1;
+	end
+	return 0;
+end
+
+-- A data structure that stores results for prime numbers as a way to "cache" results.
+-- This helps make isPrime run faster by knocking out calculations that were already done.
+local prime_memoization = {
+	[0] = false,
+	[1] = false,
+	[2] = true
+};
+function isPrime(integer, divisor)
+	--[[
+	Determines if an integer is a prime number or not.
+	]]
+	-- Makes the divisor parameter optional. By default, it should start at 2.
+	divisor = divisor or 2; 
+
+	if (prime_memoization[integer] ~= nil) then
+    	return prime_memoization[integer];
+  	end
+	
+	if (integer % divisor == 0) then
+    	prime_memoization[integer] = false;
+		return false;
+	elseif (divisor * divisor > integer) then
+    	prime_memoization[integer] = true;
+		return true;
+	end
+	-- Make a recursive call to try other factors.
+	return isPrime(integer, divisor + 1); 
+end
+
+function inverseLerp(initial, final, value)
+	--[[
+	Inverse linear interpolation function. Takes two number values, initial
+	and final, and then a third number "value". Returns the percentage of
+	"value" between initial and final.
+	]]
+	return (value - initial) / (final - initial);
+end
+
+function lerp(initial, final, i) 
+	--[[
+	Linear interpolation function. Takes number value initial and returns a
+	value that is (i * 100) percent to final value.
+	
+	i is a number value between 0 and 1
+	
+	Thanks to Zly for showing me the "precise" method
+	]]
+	return (1 - i) * initial + i * final;
+end
+
+function printTable(luaTable, whitespace)
+	--[[
+	Prints a table to the console. This includes all key value pairs, and recursively prints out any 
+	tables of additional dimensions
+
+	Formatting of table contents are done with the "whitespace" character. Courtesy of Zly.
+	]]
+	whitespace = whitespace or "";
+	
+	if (type(luaTable) ~= "table") then return end;
+	
+	for k, v in pairs(luaTable) do
+		print(whitespace..tostring(k), v);
+		if (type(v) == "table" and v ~= luaTable) then
+			tablePrint(v, whitespace.."\t");
+		end
+	end
+end 
+
+function randomFloat(minimum, maximum)
+	--[[
+	Returns a random value between minimum and maximum, but is a floating point
+	number.
+	]]
+	return math.random() * (maximum - minimum) + minimum;
+end
+
+function roundFloat(floatingNumber)
+	--[[
+	Takes in a number value and follows true rounding rules.
+	Decimals < .5 will round down, while >= .5 will round up.
+	
+	Returns the rounded number value
+	]]
+	return math.floor(floatingNumber + 0.5);
+end
+
+function shuffle(t)
+	--[[
+		"Shuffles" an array by swapping elements randomly across a table.
+	]]
+	local j
+	for i = #t, 2, -1 do
+		j = math.random(i)
+		t[i], t[j] = t[j], t[i]
+	end
+	return t
+end
+
+function simplifyFloat(number, places) 
+	--[[
+	Takes in a number value and rounds the number to places decimal 
+	places. Very useful to avoid precision errors.
+	
+	Returns a number value (usually with a floating decimal point)
+	]]
```

Comment:
Writing comments like this is what and Pythonish styled for writing comments for methonds, which is stupid to apply here, since it's not a method, and in this language in general.
And it's just messy af lol.

Reply:
These were comments that have been in my mathExtended module, which again is because Python habits stuck to me at the time. 

But here's the thing. These are multilined comments. The format that you've given me wouldn't work so nicely with multilined commenting, especially using whitespace for additional formatting. You'll have to offer examples on how I'd be able to pull off multilined comments with your convention of function documenting.

Reply:
Is it so hard to move the comment above and use a shortcut for commenting?

Reply:
Well I don't think this looks pretty

```lua
-- Takes in a number value and rounds the number to places decimal 
-- places. Very useful to avoid precision errors.
	
-- Returns a number value (usually with a floating decimal point)
function simplifyFloat(number, places)
```

Reply:
makes more sense to me and more pretty because to me it felt like you tried to comment something inside of the function instead of function itself, so.

Also idk if your IDE/Lua Plugin supports it or whatever but I do this kind of stuff sometimes:
```lua
---Sets the scale of the *`wall`*, *`sprite`*, or *`spriteQuads`*.
    ---Pass a table reference like {x = 0, y = 0} to tie it to the propertie, or 2 arguments for `x` and `y` or 1 argument for both `x` and `y`.
    ---@vararg table[]|number
    ---@overload fun(x:number, y:number):self
    ---@overload fun(scale:number):void
    ---@return self
    function object:setScale(...)
```
It makes sense here since it's something like "Java docs" cuz they meant to be written on top of the function to document and comment on the function. They are not written inside of it, which is weird and makes less sense idk.

Reply:
Also, your way is just not consistent with the entire style of the pack ynow. Meanwhile, we follow clang for the entire project, not just for 1 script file.
That's why I nitpick, this file doesn't follow the style of the pack in general at all.

Reply:
Your method of commenting is good, but it will only work for that particular IDE. I can try and rewrite the comments to get close to it, but I don't think they need to be exactly following that format.

Reply:
I know, but again it's just you don't follow the style of the entirety of the pack scripts and such, and it inconsistent for the whole project.

Reply:
So logically you will have to change everything to this style.